### PR TITLE
Prevent duplicate loads when reload is triggered

### PR DESCRIPTION
**Pull request by nathanstitt**
This was discovered by new code in `TeacherTaskPlanListing` which has a `LoadableItem` for the dashboard. The Dashboard's props are updated quite a few times (300+) when it's loaded directly. 

`LoadableItem` in turn has a `componentDidUpdate` that calls it's `reload` method every time, which blindly fires a load request.  This adds a check to only fire the load if it's not already loading.

I **think** this is safe but should be reviewed carefully since it does change the behavior of `LoadableItem`, which is used pervasively throughout our code-base.

Intended to prevent shenanigans like this:

screen shot 2016-06-28 at 10 11 29 am

### CODE REVIEW

**Comment by philschatz** [comment]
Nice! Is this indicative of a bigger bug? Like why would the screen be re-processing (recreating the VirtualDOM) hundreds of times per second?

**Comment by pandafulmanda** [comment]
i dont think this happens anywhere but on local right? but looks like a good fix either way

**Comment by nathanstitt** [comment]
I'm not sure if it's a local-only bug or not.  Would like to test on -dev, but it's currently down :(

File: src/components/loadable-item.cjsx [hunk 1]
```diff
@@ -31,15 +31,16 @@ module.exports = React.createClass
...
     return unless id?
 
     # Skip reloading if all the props are the same (the case in the Calendar for some reason)
     return if oldProps.id is id and oldProps.store is store and oldProps.actions is actions and
       oldProps.load is load and _.isEqual(oldProps.options, options)
 
     load ?= actions.load
-    unless store.isNew(id, options)
+    isLoading ?= store.isLoading
+    unless store.isNew(id, options) and not isLoading?(id, options)
```
Comment (pandafulmanda) [inline src/components/loadable-item.cjsx]: hmmm...i see how this is the same but would `unless isLoading?(id, options) or store.isNew(id, options)` be more readable?

Comment (nathanstitt) [inline src/components/loadable-item.cjsx]: great point.  your version does read better.  I've updated to do that

Comment (nathanstitt) [inline src/components/loadable-item.cjsx]: Yeah... so I tried that and it didn't work out well.  there's code in the task plans that check if the stores are reading & homework specs that expect the store to load unless it's new.  My moving the isLoading condition first the isNew status isn't checked.  

I could fixup the specs but am worried that there might be other untested areas that have the same logic.  Probably best to leave as is despite the better readability of your version

**Comment by nathanstitt** [comment]
Closing in favor of #1139 which fixes this issue by doing the check in api.coffee

**Pull request closed without merging**